test/github_pullrequest_concurrency_test.go

```diff
@@ -49,6 +56,40 @@ func TestGithubSecondPullRequestConcurrency1by1WithError(t *testing.T) {
 	})
 }
 
+func TestGithubGlobalRepoConcurrencyLimit(t *testing.T) {
+	ctx := context.Background()
+	label := "Github PullRequest Concurrent Two at a Time Set by Global Repo"
+	// create global repo
+	ctx, globalNS, runcnx, err := createGlobalRepo(ctx)
+	assert.NilError(t, err)
+	defer (func() {
+		err = cleanUpGlobalRepo(runcnx, globalNS)
+		assert.NilError(t, err)
+	})()
+
+	numberOfPipelineRuns := 10
+	// in this test we don't want to have `concurrency_limit` in local repo because
+	// global repository's `concurrency_limit` is being tested here.
+	maxNumberOfConcurrentPipelineRuns := -1
```

Comment:
Can we rename to different name as its meaning is confusing when concurrency is set by globalRepo

How about 

`localRepoMaxConcurrentRuns` or `maxConcurrentPipelineRunsLocalRepo`